packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -406,68 +406,91 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
+                    let retrievedSnapshot: ISnapshotContents | undefined;
+                    // Here's the logic to grab the persistent cache snapshot implemented by the host
+                    // Epoch tracker is responsible for communicating with the persistent cache, handling epochs and cache versions
                     const cachedSnapshotP: Promise<ISnapshotContents | undefined> =
                         this.epochTracker.get(createCacheSnapshotKey(this.odspResolvedUrl))
-                            .then((snapshotCachedEntry: ISnapshotCachedEntry) => {
+                            .then(async (snapshotCachedEntry: ISnapshotCachedEntry) => {
                                 if (snapshotCachedEntry !== undefined) {
                                     // If the cached entry does not contain the entry time, then assign it a default of 30 days old.
-                                    // eslint-disable-next-line @typescript-eslint/dot-notation
-                                    props["cacheEntryAge"] = Date.now() - (snapshotCachedEntry.cacheEntryTime ??
+                                    const age = Date.now() - (snapshotCachedEntry.cacheEntryTime ??
                                         (Date.now() - 30 * 24 * 60 * 60 * 1000));
+
+                                    // Record the cache age
+                                    // eslint-disable-next-line @typescript-eslint/dot-notation
+                                    props["cacheEntryAge"] = age;
+
+                                    // TODO: Refactor this logic (maybe move to a separate function)
+                                    // TODO: Currently there is a max of two days. Potentially remove this logic once proactive expiry exists.
+                                    // Set the max age to 2 days, a network call to retrieve the snapshot will be made if undefined is returned.
+                                    const twoDays = 2 * 24 * 60 * 60 * 1000;
+                                    const policyMaxCacheTime = hostSnapshotOptions?.maxSessionActiveTime;
+                                    const maxCacheAge = policyMaxCacheTime === undefined || policyMaxCacheTime > twoDays ? twoDays : policyMaxCacheTime;
+                                    if(age > maxCacheAge) {
+                                        await this.epochTracker.removeEntries();
+                                        this.logger.sendTelemetryEvent({ eventName: "CacheExpired", duration: age });
+                                        return undefined;
+                                    }
                                 }
+
                                 return snapshotCachedEntry;
                         });
 
+                    // Setup the network snapshot call
+                    const networkSnapshotP = this.fetchSnapshot(hostSnapshotOptions);
+
+                    // Either retrieve the both the network and cache snapshots concurrently and pick the first to return
```

Comment:
nit: extra the.

Also, consider adding that one this decision is made based on the concurrentSnapshotFetch policy.